Panic on images with zero layers

Stereoscope panics when returning results for an image that has no layers (e.g. scratch). This can be seen by running the following from stereoscope's root directory:

```
skopeo --override-os linux copy docker://anchore/test_images:scratch_v1 oci:./test-scratch
go run ./examples/basic.go oci-dir:./test-scratch
```

The cause of this panic is here: https://github.com/anchore/stereoscope/blob/main/pkg/image/image.go#L200

This code assumes there is at least one layer in the image and thus doesn't check to make sure `len(i.Layers)-1` produces a valid index.

**NOTE:** If you try to use Docker Engine to save the image you'll get a _totally different_ error, as seen here:

```
go run ./examples/basic.go anchore/test_images:scratch_v1
panic: unable to save image tar: Error response from daemon: empty export - not implemented
```

--- a/go.mod
+++ b/go.mod
@@ -23,6 +23,7 @@ require (
 	github.com/sergi/go-diff v1.1.0
 	github.com/spf13/afero v1.2.2
 	github.com/stretchr/testify v1.6.0 // indirect
+	github.com/vbatts/tar-split v0.11.1 // indirect
 	github.com/wagoodman/go-partybus v0.0.0-20200526224238-eb215533f07d
 	github.com/wagoodman/go-progress v0.0.0-20200621122631-1a2120f0695a
 	golang.org/x/net v0.0.0-20200602114024-627f9648deb9 // indirect

--- a/pkg/image/image.go
+++ b/pkg/image/image.go
@@ -197,7 +197,14 @@ func (i *Image) squash(prog *progress.Manual) error {
 
 // SquashedTree returns the pre-computed image squash file tree.
 func (i *Image) SquashedTree() *filetree.FileTree {
-	return i.Layers[len(i.Layers)-1].SquashedTree
+	layerCount := len(i.Layers)
+
+	if layerCount == 0 {
+		return filetree.NewFileTree()
+	}
+
+	topLayer := i.Layers[layerCount-1]
+	return topLayer.SquashedTree
 }
 
 // FileContentsFromSquash fetches file contents for a single path, relative to the image squash tree.

--- a/pkg/image/image_test.go
+++ b/pkg/image/image_test.go
@@ -3,11 +3,12 @@ package image
 import (
 	"crypto/sha256"
 	"fmt"
-	"github.com/go-test/deep"
-	"github.com/google/go-containerregistry/pkg/name"
 	"io/ioutil"
 	"os"
 	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/google/go-containerregistry/pkg/name"
 )
 
 func TestImageAdditionalMetadata(t *testing.T) {
@@ -95,4 +96,25 @@ func TestImageAdditionalMetadata(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestImage_SquashedTree(t *testing.T) {
+	t.Run("zero layers", func(t *testing.T) {
+		i := Image{
+			Layers: []*Layer{},
+		}
+
+		defer func() {
+			if r := recover(); r != nil {
+				t.Errorf("panicked (and recovered) while computing squashed tree for image with zero layers: %v", r)
+			}
+		}()
+
+		// Asserting that this call doesn't panic (regression: https://github.com/anchore/stereoscope/issues/56)
+		result := i.SquashedTree()
+
+		if result == nil {
+			t.Error("expected an initialized, empty FileTree, but got a nil FileTree")
+		}
+	})
 }

